File: hackerrank/easy/0015-drawing-book/python/solution.py
## solution.py

def page_count(n: int, p: int) -> int:
    from_front = p // 2
    from_back = (n // 2) - (p // 2)
    return min(from_front, from_back)


# --- demo run ---
if __name__ == "__main__":
    print(page_count(6, 2))  # 1
    print(page_count(5, 4))  # 0

File: hackerrank/easy/0015-drawing-book/js/solution.mjs
// solution.mjs

export function pageCount(n, p) {
  const fromFront = Math.floor(p / 2);
  const fromBack = Math.floor(n / 2) - Math.floor(p / 2);
  return Math.min(fromFront, fromBack);
}

// --- demo run ---
console.log(pageCount(6, 2)); // 1
console.log(pageCount(5, 4)); // 0